planck-cljs/src/planck/from/cljs_bean/core.cljs

```diff
@@ -82,15 +93,12 @@
   ITransientAssociative
   (-assoc! [tcoll k v]
     (if editable?
-      (if (and (compatible-key? k prop->key)
-               (not (and recursive?
-                         (or (object? v)
-                             (array? v)))))
+      (if (snapshot? k v prop->key recursive?)
+        (-assoc! (transient (snapshot obj prop->key key->prop recursive?)) k v)
```

Comment:
This might be obvious after a more thorough exploration of the code but I presume moving this function before the `do` doesn't adversely impact things?

Reply:
In this case, the logic in the `if` test is being flipped. (In hindsight, I probably should have simply made the upstream change as

````
(if-not (snapshot? ...
```

and left the then and else branches in place.

What's going on here, from the big picture is that for certain `assoc` operations, a bean cannot be a bean after the assoc. In that case, a snapshot of the bean needs to be taken as a persistent data structure.

Reply:
Gotcha. Makes sense—ta! :)